kubernetes-cronjob/massdriver.yaml

```diff
@@ -5,71 +5,180 @@ source_url: github.com/YOUR_ORG/<md .Name md>
+  policies: []
+    # Use jq expressions to select policies from artifact security blocks
+    # - .connections.sqs.data.security.iam.subscribe
+
 params:
+  required:
+    - namespace
+    - image
+    - resourceRequests
+    - job
   properties:
-    name:
-      type: string
     namespace:
-      type: string
+      title: Kubernetes Namespace
+      description: Application will be deployed into this namespace. If the namespace doesn't exist, it will be created.
+      $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/k8s-resource-name.json
       default: default
     image:
+      title: Container Image
       type: object
+      required:
+        - repository
+        - tag
       properties:
         repository:
-          type: string
-        sha:
+          $md.immutable: true
+          title: Repository
+          # Container repositories can be created or imported into massdriver at: https://app.massdriver.cloud/container-repositories
+          # You'll generally want to hard code your container repository in your bundle code, but it can be helpful to show it
+          # in your schema as well.
+          # If you need to allow an end-user developer to select the container repo, you can use our drop down widget documented here: https://docs.massdriver.cloud/bundles/custom-widgets-and-fields
+          description: Container repository to use for this application.
           type: string
         tag:
-          type: string
-    resource_requests:
-      type: object
-      properties:
-        cpu:
-          type: string
-        memory:
-          type: string
-    autoscaling:
-      type: object
-      properties:
-        enabled:
-          type: boolean
-          default: false
-        minReplicas:
-          type: integer
-          default: 1
-        maxReplicas:
-          type: integer
-          default: 2
-        targetCPUUtilizationPercentage:
-          type: integer
-          default: 90
+          title: Image Tag
+          $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/docker-image-tag.json
+          default: latest
+    command:
+      title: Command
+      description: Command to execute as the entrypoint of the container. If left unspecified, the container default will be used (referred to as 'entrypoint' in some container runtimes, such as Docker). Formatted as an array of strings so that 'exec' form is used.
+      type: array
+      items:
+        type: string
+    args:
+      title: Arguments
+      description: Arguments to pass to the entrypoint command of the container. If left unspecified, the container default will be used (referred to as 'command' in some container runtimes, such as Docker). Formatted as an array of strings so that 'exec' form is used.
+      type: array
+      items:
+        type: string
     envs:
```

Comment:
A note on these environment variables - specifically with Helm.

This is a weird situation that makes [the ordering of values passed to the helm resource](https://github.com/massdriver-cloud/terraform-modules/blob/main/massdriver-application-helm/main.tf#L68-L69) **extremely** important.

The way this is currently set up, both the `params` as well as the application module output a top level `envs` field.  The params one is a list of objects (with `name` and `value`), the module is an object with key/value pairs, which we convert into a list of objects (with `name` and `value` [here](https://github.com/massdriver-cloud/terraform-modules/blob/main/massdriver-application-helm/main.tf#L9)).  The module's list is a **merged** list of both the UI envs as well as the `massdriver.yaml` envs, whereas the params list is just the UI envs (from [this PR](https://github.com/massdriver-cloud/terraform-modules/pull/47)).

What's important here is that [Helm does **not** merge lists in the event that they are specified multiple times](https://github.com/helm/helm/issues/3486#issuecomment-364564716).  Maps are merged, lists are not. It will take the last version of the list that was specified (which from the Helm module is the locals from the Application module which comes from a merged map).